plugin/rewrite/README.md

```diff
@@ -96,3 +96,44 @@ rewrite edns0 subnet set 24 56
 
 * If the query has source IP as IPv4, the first 24 bits in the IP will be the network subnet.
 * If the query has source IP as IPv6, the first 56 bits in the IP will be the network subnet.
+
+### Name Field Rewrites
+
+The `rewrite` plugin offers the ability to match on the name in the question section of
+a DNS request. The match could be exact, partial, or based on a regular expression
+
+The syntax for the name re-writing is as follows:
+
+```
+rewrite [continue|stop] name [exact|partial|regex] STRING STRING
+```
+
+The match type, i.e. `exact|partial|regex`, triggers re-write: 
+
+* **exact**: on exact match of the name in the question section of a request
+* **partial**: on a partial match of the name in the question section of a request
```

Comment:
Probably want `prefix` and `suffix` too, as they should be a lot faster than a general partial match (you can test and make sure).

Reply:
Will add. Planning to use TrimLeft and TrimRight functions for HasPrefix and HasSuffix functions.